carts/views.py

```diff
@@ -0,0 +1,163 @@
+            
+            obj, create     = Cart.objects.get_or_create(
+                user_id     =  data['user_id'] , 
+                product_id  = data['product_options_id']
+                )
+            
+            result = dict()
+            
+            if not create:
+                obj.quantity = F('quantity') + 1
+                obj.save()
+                result['result'] = "SUCCESS_UPDATE"
+                return JsonResponse(result,status= 201)
+            
+            item = Cart.objects.filter(product_id = data['product_options_id']).select_related('product').select_related('product').get()
+            
+            product = {
+                "product_options_id" : item.product.id,
+                "product_name"       : item.product.product.name,
+                "product_size"       : item.product.size,
+                "product_count"      : item.quantity,
+                "product_price"      : int(item.product.price)
+            }
+            
+            result["product"] = product
+            result["result"] = "SUCCESS"
+            
+            return JsonResponse(result,status= 201)
+        
+        except KeyError :
+            return JsonResponse({"message" : "KEY_ERROR"},status= 400)
+        except AttributeError :
+            return JsonResponse({"message" : "SERVER_ERROR"},status= 500)
+
+class UpdateCartView (View) :
+    @validate_quantity
+    def patch(self, request, user_id, product_id) :
+        try :
+            user = user_id
+            product = product_id
+            data            = json.loads(request.body)['message']
+            
+            obj, created    = Cart.objects.update_or_create( 
+                        user_id     = user, 
+                        product_id  =product
+            )
+
+            result = dict()
+            result["product_count"] = data["product_count"]
+            
+            if not created:
+                obj.quantity        = data["product_count"]
+                obj.save()
+                
+                result["result"]    = "SUCCEESS"
+                return JsonResponse(result,status= 201)
+            
+            result["result"]        = "SUCCEESS"
+            return JsonResponse(result,status= 201)
+        
+        except KeyError :
+            return JsonResponse({"message" : "KEY_ERROR"},status= 400)
+
+class DeleteCartView (View) :
+    
+    @validate_cartdata
+    def post(self, request) :
+        try :
+            data = json.loads(request.body)['message']
+            success  = Cart.objects.filter(
+                user_id=data['user_id'],
+                product_id=data['product_options_id']
+            ).delete()
+            if success == 0 :
+                return JsonResponse({"message" : "DELETE_FAIL"}, status= 400)
+            return JsonResponse({"message" : "DELETE_SUCCESS"}, status= 400)
+        except KeyError :
+            return JsonResponse({"message" : "KEY_ERROR"}, status= 400)
```

Comment:
RESTful하게 API를 만들기 위해서는 자원을 지울때, delete할 때, path parameter로 cart_id를 받아서 지우도록 만들어주세요.
다음과 같이 URI만 보고도 해당 자원에 어떤 작업을 수행하는지 알 수 있어야 합니다.~ `DELETE :8000/carts/1`

```suggestion
class DeleteCartView (View) :
    
    @validate_cartdata
    def post(self, request) :
        try :
            data = json.loads(request.body)['message']
            success  = Cart.objects.filter(
                user_id=data['user_id'],
                product_id=data['product_options_id']
            ).delete()
            if success == 0 :
                return JsonResponse({"message" : "DELETE_FAIL"}, status= 400)
            return JsonResponse({"message" : "DELETE_SUCCESS"}, status= 400)
        except KeyError :
            return JsonResponse({"message" : "KEY_ERROR"}, status= 400)
```